tests/phpunit/tests/theme/wpThemeHasThemeJson.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+require_once __DIR__ . '/base.php';
+
+/**
+ * Tests wp_theme_has_theme_json().
+ *
+ * @group theme_json
+ *
+ * @covers ::wp_theme_has_theme_json
+ */
+class Tests_Theme_WpThemeHasThemeJson extends WP_Theme_UnitTestCase {
+
+	/**
+	 * Test that it reports correctly themes that have a theme.json.
+	 *
+	 * @covers wp_theme_has_theme_json
+	 */
+	public function test_theme_has_theme_json() {
+		switch_theme( 'block-theme' );
+		$this->assertTrue( wp_theme_has_theme_json() );
+	}
```

Comment:
This should use a data provider to merge this test method with:
- `test_theme_has_no_theme_json()`
- `test_child_theme_has_theme_json()`
- `test_child_theme_has_not_theme_json_but_parent_has()`
- `test_neither_child_or_parent_themes_have_theme_json()`

```suggestion
	 *
	 * @dataProvider data_theme_has_theme_json_reports_correctly
	 *
	 * @param string $theme    The slug of the theme to switch to.
	 * @param bool   $expected The expected result.
	 */
	public function test_theme_has_theme_json_reports_correctly(  $theme, $expected ) {
		switch_theme( $theme );
		$this->assertSame( $expected, wp_theme_has_theme_json() );
	}
	
	/**
	 * Data provider for test_theme_has_theme_json_reports_correctly().
	 *
	 * @return array[]
	 */
	public function data_theme_has_theme_json_reports_correctly() {
		return array(
			'a theme with theme.json'       => array(
				'theme'    => 'block-theme',
				'expected' => true,
			),
			'a theme without theme.json'    => array(
				'theme'    => 'default',
				'expected' => false,
			),
			'a child theme with theme.json' => array(
				'theme'    => 'block-theme-child',
				'expected' => true,
			),
			'a child theme without theme.json and parent has theme.json' => array(
				'theme'    => 'block-theme-child-no-theme-json',
				'expected' => true,
			),
			'a child theme without theme.json and parent theme without theme.json' => array(
				'theme'    => 'default-child-no-theme-json',
				'expected' => false,
			),
		);
	}
```
Output when running `phpunit --filter test_theme_has_theme_json_reports_correctly --testdox`:

![image](https://user-images.githubusercontent.com/79332690/207469715-f267e493-10b1-41cd-b50d-ba468d7e1b32.png)